hw11/Lesson11.py

```diff
@@ -0,0 +1,64 @@
+import datetime
+import inspect
+
+
+def my_decorator_func(func):
+    def deco_func(*args, **kwargs):
+        func_name = func.__name__
+        res = func(*args, **kwargs)
+        now_ = datetime.datetime.now()
+        print(f'The function name is {func_name} - was called at {now_.strftime("%H:%M")}\n')
+        return res
+    return deco_func
+
+
+@my_decorator_func
+def my_function():
+    print('This is my func')
+
+
+my_function()
+
+
+class ContextManager:
+    def __init__(self):
+        print('=' * 10)
+        print('What is the sense of this task?')
```

Comment:
The sense of this task is to understand how context managers work and what you can do (and how) when you write your own context managers. Basically, you have 2 main points to manipulate the execution: enter, where you can set up some pre-conditions, and exit, where you can handle after-conditions. In this case, you messed a little:
1. If you don't need to use a value from the context manager you can skip the init method and remove it from here.
2. You should print the first line of = in the enter method since that method in entrypoint for your manager
3. The exit method should print 10 times of = in any case and not only when an error has occurred. To fix this, move print to the end (but not inside the `if` block) and move `return` outside the `if` block but after the print.